### PR TITLE
Fix BigInt primary key routing

### LINKED ISSUE #20288: BigInt primary key routing in the App
### Describe the Bug

When using a BigInt primary key for a collection the App will fail to route these IDs when they go beyond regular numbers which is exactly what you would use a BigInt for.

https://github.com/directus/directus/assets/9389634/cb5d0881-52e2-43b4-a4b0-dd529fff3441

### To Reproduce

- create a collection with a BigInt primary key
- create some records with IDs `> 9007199254740991` or `< -9007199254740991`
- try to open the detail page

### Directus Version

10.7.1

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by wasimTQ**
## Scope

What's changed:

- Fixes the routing error when opening a bigint value

## Potential Risks / Drawbacks

- 

## Review Notes / Questions

- I'd like to know whether it fixes the error and are there any other aspects get affected by this change

---

Fixes #20288

### CODE REVIEW

**Comment by wasimTQ** [comment]
@br41nslug Can you review it?

**Review by paescuj — COMMENTED** [review]
Thanks @wasimTQ!

The fix looks sensible 👍 (Although, I'm not able to reproduce the issue #20288 in particular).
We need to add a new test case for this before we can merge.

**Review by br41nslug — APPROVED** [review]
Using the native BigInt type solves the issue nicely on my end :D

**Review by paescuj — COMMENTED** [review]
Can confirm as well 🚀 
I'd still like to have this covered by a test.

**Comment by wasimTQ** [comment]
>  (Although, I'm not able to reproduce the issue #20288 in particular). 

@paescuj I think it's because of this issue #20287. I have a working solution of it tho. I'm currently working on that atm.

**Comment by wasimTQ** [comment]
> Can confirm as well 🚀 I'd still like to have this covered by a test.

What should the test cover? Just that the BigInt field is returning in the whereClause and the number is not changed?
I'll work on that. I'm still kinda weak in the testing aspect. So I might need some help there.

**Comment by br41nslug** [comment]
@paescuj I would love to see tests for this too however there is unfortunately a reason there are no tests for this currently 😬 This specific logic is hidden away in 3 levels deep of nested function which makes testing this separately very difficult if not impossible without refactoring
image
https://github.com/directus/directus/blob/dc3068453d10d8564606dc9b30fe82d2273b5d3a/api/src/utils/apply-query.ts#L647

**Comment by paescuj** [comment]
> @paescuj I think it's because of this issue #20287. I have a working solution of it tho. I'm currently working on that atm.

Nope, I've inserted the value directly into the DB, but apparently it needs significantly larger than max safe.
That being said, I've noticed #20287 too, so I'm glad to hear you're working on that ❤️

**Comment by paescuj** [comment]
> @paescuj I would love to see tests for this too however there is unfortunately a reason there are no tests for this currently 😬 This specific logic is hidden away in 3 levels deep of nested function which makes testing this separately very difficult if not impossible without refactoring

Hmm, I would have thought it to be feasible by passing in a corresponding schema
https://github.com/directus/directus/blob/dc3068453d10d8564606dc9b30fe82d2273b5d3a/api/src/utils/apply-query.test.ts#L7 and root filter
https://github.com/directus/directus/blob/dc3068453d10d8564606dc9b30fe82d2273b5d3a/api/src/utils/apply-query.test.ts#L126

**Comment by br41nslug** [comment]
In that case I hope im wrong 😄

**Comment by wasimTQ** [comment]
@paescuj Was the test I wrote good?

**Comment by paescuj** [comment]
> @paescuj Was the test I wrote good?

Yes, thank you! ❤️

File: .changeset/chatty-eagles-fix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/api': patch
+---
+
+Fixed routing issues happened when the primary key is more than javascript safe integer
```
Comment (paescuj) [inline .changeset/chatty-eagles-fix.md:5]:
```suggestion
Ensured filter values higher than JavaScript max safe integer work with big integer fields
```

**Review by paescuj — APPROVED** [review]

**Pull request merged**